app/javascript/components/visualization/StudyViolinPlot.js

```diff
@@ -27,29 +30,31 @@ export default function StudyViolinPlot({ study, gene }) {
     let results
     if (!clusterParams) {
       // this is the initial load
-      results = await fetchExpressionViolin(study.accession, gene)
+      results = await fetchExpressionViolin(study.accession, genes,null,null,null,null,null,collapseBy)
       setAnnotationList(results.annotation_list)
     } else {
       results = await fetchExpressionViolin(study.accession,
-        gene,
+        genes,
         clusterParams.cluster,
         clusterParams.annotation.name,
         clusterParams.annotation.type,
         clusterParams.annotation.scope,
-        clusterParams.subsample)
+        clusterParams.subsample,
+        collapseBy)
     }
     setIsLoaded(true)
     setIsLoading(false)
+    setStudyGeneNames(results.gene_names)
     renderViolinPlot(graphElementId, results)
   }
 
+  // do a load from the server if study, gene, or collapse parameter has changed
   useEffect(() => {
-    // do a load from the server if this is the initial load
-    if (!isLoading && !isLoaded) {
+    if (!isLoading) {
       loadData()
     }
-  }, [study.accession, gene])
-
+  }, [study.accession, genes[0], collapseBy])
+  const isCollapsedView = ['mean', 'median'].indexOf(collapseBy) >= 0
```

Comment:
We have many pre-existing IE incompatibilities: [arrow functions](https://caniuse.com/?search=arrow), [`fetch`](https://caniuse.com/?search=fetch), and likely a dozen or so more.  

In other words, this would not make us become incompatibility with IE.  That happened long ago!